core/src/gl/dynamicQuadMesh.cpp

```diff
@@ -1,41 +1,28 @@
 #include "gl/dynamicQuadMesh.h"
+#include "gl/error.h"
 #include "gl/renderState.h"
 #include "gl/shaderProgram.h"
 
 namespace Tangram {
 
 GLuint QuadIndices::quadIndexBuffer = 0;
 int QuadIndices::quadGeneration = -1;
-std::atomic<int> QuadIndices::meshCounter(0);
 
+void QuadIndices::load(RenderState& rs) {
 
-void QuadIndices::ref() {
-    meshCounter++;
-}
-
-void QuadIndices::unref() {
-    meshCounter--;
-
-    if (quadIndexBuffer != 0 && (!RenderState::isValidGeneration(quadGeneration) ||
-                                 meshCounter <= 0)) {
-
-        if (RenderState::indexBuffer.compare(quadIndexBuffer)) {
-            RenderState::indexBuffer.init(0, false);
-        }
+    if (quadIndexBuffer != 0 && !rs.isValidGeneration(quadGeneration)) {
```

Comment:
Ahh.. it might be too late to unset the quadIndexBuffer here, maybe it make sense to put this in the constructor of `DyanmicQuadMesh`.